src/test/java/gift/controller/WishControllerTest.java

```diff
@@ -0,0 +1,99 @@
+package gift.controller;
+
+import gift.dto.*;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.web.server.LocalServerPort;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestClient;
+
+import java.util.List;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+public class WishControllerTest {
+    private final RestClient restClient = RestClient.builder().build();
+
+    @LocalServerPort
+    private int port;
+    private String baseUrl;
+
+    @BeforeEach
+    void setBaseUrl() {
+        baseUrl = "http://localhost:"+ port;
+    }
+
+    @Test
+    void Wish_추가_테스트() {
+        String token = registerMemberAndGetToken("demo1@email", "asdf");
+
+        ResponseEntity<String> responseEntity = addWish(token, 1L, 234);
+
+        assertThat(responseEntity.getStatusCode()).isEqualTo(HttpStatus.CREATED);
+    }
+
+    @Test
+    void Wish_수정_테스트() {
+        String token = registerMemberAndGetToken("demo2@email", "asdf");
+        addWish(token, 1L, 123);
+
+        ResponseEntity<String> responseEntity = updateWish(token, 1L, 321);
+
+        assertThat(responseEntity.getStatusCode()).isEqualTo(HttpStatus.OK);
+    }
+
+    @Test
+    void Wish_GetList_테스트() {
+        String token = registerMemberAndGetToken("demo3@email", "asdf");
+        addWish(token, 1L, 123);
+
+        ResponseEntity<WishesResponseDto> responseEntity = getWishList(token);
+
+        assertThat(responseEntity.getBody().wishes()).isNotNull();
+        assertThat(responseEntity.getBody().wishes().size()).isGreaterThan(0);
+        assertThat(responseEntity.getBody().wishes().getLast().count()).isEqualTo(123);
+        assertThat(responseEntity.getBody().wishes().getLast().productResponseDto().id()).isEqualTo(1);
```

Comment:
```suggestion
        assertAll(
            () -> assertThat(responseEntity.getBody().wishes()).isNotNull(),
            () -> assertThat(responseEntity.getBody().wishes().size()).isGreaterThan(0),
            () -> assertThat(responseEntity.getBody().wishes().getLast().count()).isEqualTo(123),
            () -> assertThat(responseEntity.getBody().wishes().getLast().productResponseDto().id()).isEqualTo(1)       
        )
```

이렇게 assertAll()로 묶으면 어느 장단점이 있을까요?

Reply:
장점은 앞에 assert가 실패해도 뒤 assert까지 검사할 수 있는 것 같습니다.
단점은 코드가 비대해지는 것 같습니다!

이렇게 수정해보겠습니다